envpool/mujoco/dmc/cheetah.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 Garena Online Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ENVPOOL_MUJOCO_DMC_CHEETAH_H_
+#define ENVPOOL_MUJOCO_DMC_CHEETAH_H_
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+#include <random>
+#include <string>
+
+#include "envpool/core/async_envpool.h"
+#include "envpool/core/env.h"
+#include "envpool/mujoco/dmc/mujoco_env.h"
+#include "envpool/mujoco/dmc/utils.h"
+
+namespace mujoco_dmc {
+
+std::string GetCheetahXML(const std::string& base_path,
+                          const std::string& task_name) {
+  return GetFileContent(base_path, "cheetah.xml");
+}
+
+class CheetahEnvFns {
+ public:
+  static decltype(auto) DefaultConfig() {
+    return MakeDict("max_episode_steps"_.Bind(1000), "frame_skip"_.Bind(4),
+                    "task_name"_.Bind(std::string("run")));
+  }
+  template <typename Config>
+  static decltype(auto) StateSpec(const Config& conf) {
+    return MakeDict("obs:position"_.Bind(Spec<mjtNum>({8})),
+                    "obs:velocity"_.Bind(Spec<mjtNum>({9}))),
+           "discount"_.Bind(Spec<float>({-1}, {0.0, 1.0}));
+  }
+  template <typename Config>
+  static decltype(auto) ActionSpec(const Config& conf) {
+    return MakeDict("action"_.Bind(Spec<mjtNum>({-1, 6}, {-1.0, 1.0})));
+  }
+};
+
+using CheetahEnvSpec = EnvSpec<CheetahEnvFns>;
+
+// https://github.com/deepmind/dm_control/blob/1.0.2/dm_control/suite/cheetah.py#L60
+class CheetahEnv : public Env<CheetahEnvSpec>, public MujocoEnv {
+ protected:
+  const mjtNum kRunSpeed = 10;
+  std::uniform_real_distribution<> dist_uniform_;
+
+ public:
+  CheetahEnv(const Spec& spec, int env_id)
+      : dist_uniform_(0, 1),
+        Env<CheetahEnvSpec>(spec, env_id),
+        MujocoEnv(
+            spec.config["base_path"_],
+            GetCheetahXML(spec.config["base_path"_], spec.config["task_name"_]),
+            spec.config["frame_skip"_], spec.config["max_episode_steps"_]) {}
+
+  void TaskInitializeEpisode() override {
+    assert(model_->njnt == model_->nq);
+    int is_limited = static_cast<int>(*(model_->jnt_limited)) == 1;
```

Comment:
is_limited is a bool array instead of int. Please check the original code.
```python
is_limited
[False False False  True  True  True  True  True  True]
physics.model.jnt_range
[[ 0.          0.        ]
 [ 0.          0.        ]
 [ 0.          0.        ]
 [-0.52359878  1.04719755]
 [-0.87266463  0.87266463]
 [-4.01425728  0.87266463]
 [-0.99483767  0.00698132]
 [-1.22173048  0.87266463]
 [-0.48869219  0.48869219]]
physics.model.jnt_range[is_limited].T
[[-0.52359878 -0.87266463 -4.01425728 -0.99483767 -1.22173048 -0.48869219]
 [ 1.04719755  0.87266463  0.87266463  0.00698132  0.87266463  0.48869219]]
```